### ISSUE TITLE
Module Support  |  Decision

### ISSUE THREAD

**Issue by Alkali-Metal** [Adventurey-Things/Adventure-Bot#11]
Okay, so, modules, we need to integrate them into the adventure code. There needs to be a few things decided first though:
- We need the module code in a separate place from everything else. I suggest maybe a `modules/` directory.
- We might want to have each module use a separate database than the main one along with any other data it needs. I was thinking something along the lines of `modules/<ModuleName>/` for the data that each module uses/needs.
- We need a way to tell the main code in the `adventure.py` that it needs to be sent to a specific module. I was thinking we might be able to use the base `"to"` keys but have only one key where it is `"module:<ModuleName>":{}` and then have everything that the module needs in that dictionary.

Actually give your input on this Quirky, it's a major thing.

**Comment by MrLotU** [comment]
So, with modules do you mean like a thing (flag or something) that we (adventure creators) can implement? (example: The timer thing Lina talked about)

**Comment by Alkali-Metal** [comment]
Yeah, so essentially how we were thinking of doing modules would be by having something like `"MODULENAME:location_alias":{}` then the module would handle all of the stuff in that location and the main code would just send all of that data to the module. Although that would only allow a single module to be used per adventure, which isn't really optimal by any means.

**Comment by MrLotU** [comment]
Why would that only leave for 1 module per thing though

**Comment by Alkali-Metal** [comment]
Well... with the way I explained it, each location_alias could only use `"modulename:"` modules. OHWAIT!

we could do it like
```json
"location_alias":{
  "modules":{
    "<ModuleName>":{
      <Data for module in here>
    },
  }
}
```
and that would of course hold the data that we send to "ModuleName" and then we could send the required data to the corresponding module with support for multiple modules.
